src/Build/BackEnd/Components/Logging/EventSourceSink.cs

```diff
@@ -970,6 +988,75 @@ private void RaiseTelemetryEvent(object sender, TelemetryEventArgs buildEvent)
             }
         }
 
+        /// <summary>
+        /// Determines if the specified warning should be treated as an error.
+        /// </summary>
+        /// <param name="warningEvent">A <see cref="BuildWarningEventArgs"/> that specifies the warning.</param>
+        /// <returns><code>true</code> if the warning should be treated as an error, otherwise <code>false</code>.</returns>
+        private bool ShouldTreatWarningAsError(BuildWarningEventArgs warningEvent)
+        {
+            // This only applies if the user specified /warnaserror from the command-line or added an empty set through the object model
+            //
+            if (WarningsAsErrors != null)
```

Comment:
The name is technically part of our public API already as a property to `BuildParameters`.  I could rename the internal property on `EventSourceSink` if you feel strongly about it.